### ISSUE TITLE
.netrc support

### ISSUE THREAD

**Issue by merdely** [restic/restic#3446]
Output of `restic version`
--------------------------

restic 0.12.0 compiled with go1.15.8 on linux/amd64

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

Instead of requiring username/password be passed as part of the repository for using rest-server, use .netrc.  I run restic from cron and reports get emailed to me.  If there are any issues, the password my restic client uses to connect to my rest-server is in the email (sent to Gmail).  If restic were to use .netrc for web authentication, that would not happen.

What are you trying to do? What problem would this solve?
---------------------------------------------------------

Prevent credential leaking.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Absolutely.  Restic is a fantastic backup tool.  I recently had to rebuild my laptop (I use Arch, btw).  And I was able to use restic to restore my files easily using `restic mount -H mylaptop /mnt`.  Thank you for restic and rest-server.

**Comment by MichaelEischer** [comment]
Please provide an example log output which shows where the password is contained in the issue reports. Does using `--repository-file` or passing the repo URL via environment variables solve your problem?

**Comment by merdely** [comment]
I'm going to close the issue for now.  I realize that at the time I was running an older version and am now using --repository-file.  I did have a couple of restic processes that I needed to kill (due to something else going on with my system) which caused the full restic command line to show up in the email subject and with --repository-file, my http password was not there.

The issue I was referring to in the past had the http password in actual restic output (so in the body of the message), but like I said, it was an old version, so I'll see if it happens again and open a new issue.
